mitiq/rem/post_select.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from mitiq._typing import MeasurementResult
+
+
+def post_select(
+    measurement_result: MeasurementResult,
+    hamming_weight: int,
+    inverted: bool = False,
+) -> MeasurementResult:
```

Comment:
We could keep the name `post_select` for something more general that we could introduce in the future. E.g. one of the arguments could be a custom function to specify what should be post-selected e.g.  `select: Callable[[MeasurementResult], bool] `. 

The present funciton could be renamed to something more specific `post_select_weight` or  `select_hamming_weight` or  etc..

Reply:
Great suggestion!!

Reply:
Done